packages/docs-reanimated/docs/guides/migration-from-3.x.md

```diff
@@ -8,18 +8,92 @@ sidebar_label: Migration from 3.x
+import { Reanimated3DefaultSpringConfigWithDuration } from 'react-native-worklets'; // For duration based defaults.
+```
+
+You can explore the changes in the default parameters in [withSpring reference](/docs/next/animation/withSpring).
+
+### Removed `useWorkletCallback`
+
+`useWorkletCallback` was marked as deprecated in Reanimated 3 and removed in Reanimated 4.
+
+You will need to manually update all its usages in the following way:
+
+1. Add `'worklet';` directive inside the function
+2. Add dependency array according to the rules of hooks
+
+```jsx
+// highlight-next-line
+useCallback(() => {
+  // highlight-next-line
+  'worklet';
+  // some code
+  // highlight-next-line
+}, [...dependencies]);
+```
+
+If you're using [gorhom/react-native-bottom-sheet](https://github.com/gorhom/react-native-bottom-sheet), please update to latest version.
+
+### Removed `addWhitelistedNativeProps` and `addWhitelistedUIProps`
+
+Reanimated 4 removes the concept of native and UI props. `addWhitelistedNativeProps` and `addWhitelistedUIProps` are now no-ops, marked as deprecated and planned to be removed in a future release of Reanimated. You are safe to remove all usages of these functions across your codebase.
+
+### Removed `combineTransition`
+
+`combineTransition` was deprecated in Reanimated 3 and removed in Reanimated 4. You can achieve the same effect using the syntax below. For more details, check out [Entry/Exit Transition](/docs/layout-animations/layout-transitions/#entryexit-transition).
+
+```js
+EntryExitTransition.entering(entering).exiting(exiting);
+```
+
+### Removed support for `react-native-v8`
+
+Reanimated 4 removes support for [react-native-v8](https://github.com/Kudo/react-native-v8) engine. We no longer actively support V8 and the project itself seems to be abandoned as there hasn't been any new commits since August 2024.
+
+If you are affected by this change, please to leave a comment under the [related PR](https://github.com/software-mansion/react-native-reanimated/pull/7650) and explain your motivation to use V8 engine with Reanimated. We might consider restoring support for V8 engine at some point in the future.
+
+### Renamed `useScrollViewOffset` to `useScrollOffset`
+
+In Reanimated 4, we renamed `useScrollViewOffset` to `useScrollOffset`. For the sake of backwards compatibility, we decided to keep `useScrollViewOffset` for now but it is marked as deprecated and planned to be removed in a future release of Reanimated. You will need to update all `useScrollViewOffset` usages to `useScrollOffset`.
+
+### Deprecated `useAnimatedGestureHandler`
+
+`useAnimatedGestureHandler` was marked as deprecated in Reanimated 3 and will be removed in a future relase of Reanimated 4. You will need to upgrade all its usages to the new `Gesture` API introduced in Gesture Handler 2 according to the steps described in [React Native Gesture Handler documentation](https://docs.swmansion.com/react-native-gesture-handler/docs/guides/upgrading-to-2).
```

Comment:
i'd rewrite there sentences to not use future tense coz for readers it will be the present time

also typo `relese` -> `release`